### ISSUE TITLE
Add support for reusable templates -> workflows

### ISSUE THREAD

**Issue by samsmithnz** [samsmithnz/AzurePipelinesToGitHubActionsConverter#313]

**Comment by calebcartwright** [comment]
Hey Sam 👋 is this issue intended to (eventually) address this https://github.com/samsmithnz/AzurePipelinesToGitHubActionsConverter#templates or is "templates" referring to something different here?

**Comment by samsmithnz** [comment]
@calebcartwright hey there! 👋 Yes this is the right issue, but it's not straight forward, as they don't translate over directly and are handled differently in GitHub Actions. Still not sure on the best translation. Any thoughts?

**Comment by calebcartwright** [comment]
Yeah I originally came over here with the assumption it wasn't supported as it seems like a really hard thing to even attempt.

My intuition at this point would be that the only remotely feasible option for semi-converting templates would be to attempt to resolve template references and then run your own expansion engine over them to get the resultant stage/job/steps that could then be converted. However that also sounds like a royal pain!

Would be super helpful if some of your former colleagues would consider open sourcing or sharing the internals of how Azure DevOps does that today to generate the build plan 😁

**Comment by samsmithnz** [comment]
Yes - makes sense and agree. To be clear - this will convert templates very well today- but it's the reusable template part that is more challenging.

**Comment by calebcartwright** [comment]
Gotcha, thanks for the clarification. Seems like a massive undertaking but on the off chance I ever find myself with a couple months of consecutive free time...
